src/pages/teaching-material/index.tsx

```diff
@@ -65,13 +124,96 @@ export default function Discover() {
     []
   )
 
+  const getResultsSummary = useCallback(() => {
+    const dataResults = `showing ${results.courses.data.length} courses, ${results.lectures.data.length} lectures, and ${results.blocks.data.length} lecture blocks.`
+    const authorPart =
+      selectedAuthors.length !== 0
+        ? `${selectedAuthors.length} ${
+            selectedAuthors.length > 1 ? 'authors' : 'author'
+          }.`
+        : ''
+    const keywordPart =
+      selectedKeywords.length !== 0
+        ? `${selectedKeywords.map((k) => k.ariaLabel || k.label).join(', ')}`
+        : ''
+    const isFiltered = keywordPart || authorPart
+    const fullSummary = `${dataResults} ${
+      isFiltered ? 'Filtered by' : ''
+    } ${keywordPart} ${authorPart ? 'and' : ''} ${authorPart}`
+    return fullSummary
+  }, [
+    // courseResults,
+    // lectureResults,
+    // blockResults,
+    results.courses.data.length,
+    results.lectures.data.length,
+    results.blocks.data.length,
+    selectedAuthors,
+    selectedKeywords,
+  ])
+
+  const onChange = useCallback(async () => {
+    const keywords = selectedKeywords.map((keyword) => keyword.label)
+    const authors = selectedAuthors.map((author) => author.label)
+    const courseFilterPromise = filterCourseOnKeywordsAndAuthors(
+      keywords,
+      authors,
+      currentCoursePageNumber,
+      selectedSort.id,
+      DEFAULT_MATCHES_PER_PAGE
+    )
+    const lectureFilterPromise = filterLectureOnKeywordsAndAuthors({
+      keywords,
+      authors,
+      pageNumber: currentLecturePageNumber,
+      sortMethod: selectedSort.id,
+      matchesPerPage: DEFAULT_MATCHES_PER_PAGE,
+    })
+    const blockFilterPromise = filterBlockOnKeywordsAndAuthors({
+      keywords,
+      authors,
+      pageNumber: currentBlockPageNumber,
+      sortMethod: isBlockSortOptionType(selectedSort.id)
+        ? selectedSort.id
+        : 'ALPHABETICAL_ASC',
+      matchesPerPage: DEFAULT_MATCHES_PER_PAGE,
+    })
+    const [courseFilterResult, lectureFilterResult, blockFilterResult] =
+      await Promise.all([
+        courseFilterPromise,
+        lectureFilterPromise,
+        blockFilterPromise,
+      ])
+
+    setResults({
+      courses: courseFilterResult,
+      lectures: lectureFilterResult,
+      blocks: blockFilterResult,
+    })
+
+    // setLectureResults(lectureFilterResult)
+    // setCourseResults(courseFilterResult)
+    // setBlockResults(blockFilterResult)
+  }, [selectedKeywords, selectedAuthors, selectedSort])
+
+  useEffect(() => {
+    onChange()
+  }, [
+    selectedKeywords.length,
+    selectedAuthors.length,
+    selectedSort.id,
+    onChange,
+  ])
```

Comment:
Yeah, it shouldn't be possible to change e.g a keyword without first removing it (changing length) and then adding another one (changing length again), right?

Reply:
Yes, I think just these three will be enough, nothing else really changes the results or their order right now.